modules/pubsub/src/main/java/com/spotify/ffwd/pubsub/PubsubOutputPlugin.java

```diff
@@ -0,0 +1,147 @@
+import com.google.inject.Singleton;
+import com.google.inject.name.Names;
+import com.google.pubsub.v1.ProjectTopicName;
+import com.spotify.ffwd.filter.Filter;
+import com.spotify.ffwd.module.Batching;
+import com.spotify.ffwd.output.OutputPlugin;
+import com.spotify.ffwd.output.OutputPluginModule;
+import com.spotify.ffwd.output.PluginSink;
+import com.spotify.ffwd.serializer.Serializer;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Optional;
+import org.threeten.bp.Duration;
+
+public class PubsubOutputPlugin extends OutputPlugin {
+  private static final long DEFAULT_BYTES_THRESHOLD = 5000L;
+  private static final long DEFAULT_COUNT_THRESHOLD = 1000L;
+  private static final long DEFAULT_DELAY_THRESHOLD = Duration.ofMillis(100).toNanos();
+
+
+  private final Optional<Serializer> serializer;
+
+  private final Optional<String> serviceAccount;
+  private final Optional<String> project;
+  private final Optional<String> topic;
+
+  private final long requestBytesThreshold;
+  private final long messageCountBatchSize;
+  private final Optional<Long> publishDelayThresholdMs;
+
+  @JsonCreator
+  public PubsubOutputPlugin(
+      @JsonProperty("filter") Optional<Filter> filter,
+      @JsonProperty("flushInterval") Optional<Long> flushInterval,
+      @JsonProperty("batching") Optional<Batching> batching,
+      @JsonProperty("serializer") Serializer serializer,
+
+      @JsonProperty("serviceAccount") String serviceAccount,
+      @JsonProperty("project") String project,
+      @JsonProperty("topic") String topic,
+
+      @JsonProperty("requestBytesThreshold") Long requestBytesThreshold,
+      @JsonProperty("messageCountBatchSize") Long messageCountBatchSize,
+      @JsonProperty("publishDelayThresholdMs") Long publishDelayThresholdMs
+  ) {
+    super(filter, Batching.from(flushInterval, batching));
+    this.serializer = Optional.ofNullable(serializer);
+
+    this.serviceAccount = Optional.ofNullable(serviceAccount);
+    this.project = Optional.ofNullable(project);
+    this.topic = Optional.ofNullable(topic);
+
+    this.requestBytesThreshold = Optional.ofNullable(
+        requestBytesThreshold).orElse(DEFAULT_BYTES_THRESHOLD);
+    this.messageCountBatchSize = Optional.ofNullable(
+        messageCountBatchSize).orElse(DEFAULT_COUNT_THRESHOLD);
+    this.publishDelayThresholdMs = Optional.ofNullable(publishDelayThresholdMs);
+  }
+
+  @Override
+  public Module module(final Key<PluginSink> key, final String id) {
+    return new OutputPluginModule(id) {
+
+      @Provides
+      @Singleton
+      public Publisher publisher() throws IOException {
+        Preconditions.checkArgument(project.isPresent(), "Google project must be set in config");
+        Preconditions.checkArgument(topic.isPresent(), "Pubsub topic must be set in config");
+
+        final Duration publishDelayThreshold = Duration.ofNanos(
+            publishDelayThresholdMs.orElse(DEFAULT_DELAY_THRESHOLD));
```

Comment:
Perhaps push this `orElse` to the constructor as you did with `this.messageCountBatchSize`

Reply:
Fixed.